projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -492,27 +493,35 @@ public function render_block( $attributes, $content ) {
 			$wrapper_attributes = \WP_Block_Supports::get_instance()->apply_block_supports();
 		}
 
-		$headline_id     = 'relatedposts-headline-' . uniqid();
 		$headline_markup = '';
 
-		if ( $block_attributes['show_headline'] === true ) {
+		if ( isset( $block ) ) {
+			foreach ( $block->inner_blocks as $inner_block ) {
+				if ( 'core/heading' === $inner_block->name && ! empty( trim( wp_strip_all_tags( $inner_block->inner_html ) ) ) ) {
+					$headline_markup = $inner_block->inner_html;
+					break;
+				}
+			}
+		}
+
+		if ( empty( $headline_markup ) && $block_attributes['show_headline'] === true ) {
 			$headline = $block_attributes['headline'];
 			if ( strlen( trim( $headline ) ) !== 0 ) {
 				$headline_markup = sprintf(
-					'<h3 id="%2$s" class="jp-relatedposts-headline">%1$s</h3>',
-					esc_html( $headline ),
-					esc_attr( $headline_id )
+					'<h3 class="jp-relatedposts-headline">%1$s</h3>',
+					esc_html( $headline )
 				);
 			}
 		}
+
 		$display_markup = sprintf(
-			'<nav class="jp-relatedposts-i2%1$s"%2$s data-layout="%3$s" %6$s>%4$s%5$s</nav>',
+			'<nav class="jp-relatedposts-i2%1$s"%2$s data-layout="%3$s" aria-label="%6$s">%4$s%5$s</nav>',
 			! empty( $wrapper_attributes['class'] ) ? ' ' . esc_attr( $wrapper_attributes['class'] ) : '',
 			! empty( $wrapper_attributes['style'] ) ? ' style="' . esc_attr( $wrapper_attributes['style'] ) . '"' : '',
 			esc_attr( $block_attributes['layout'] ),
 			$headline_markup,
 			$list_markup,
-			empty( $headline_markup ) ? 'aria-label="' . esc_attr( __( 'Related Posts', 'jetpack' ) ) . '"' : 'aria-labelledby="' . esc_attr( $headline_id ) . '"'
```

Comment:
Is it okay that we're losing the `aria-labelledby` markup in this case?

Reply:
Yes, I should have added a note about it. My initial approach was to add an id to the heading tag and use `aria-labelledby` to reference it in the nav label. However, I only managed to get the headline markup and content as a string (with `$inner_block->inner_html`), and parsing it to add an attribute and reconstruct the string seemed overkill compared to using `aria-label` with the heading text content.